SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialReplicationGraph.cpp

```diff
@@ -63,24 +73,63 @@ void USpatialReplicationGraph::OnOwnerUpdated(AActor* Actor, AActor* OldOwner)
 
 void USpatialReplicationGraph::PreReplicateActors(UNetReplicationGraphConnection* ConnectionManager)
 {
+	USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(NetDriver);
+	FSpatialReplicationGraphLoadBalancingContext LoadBalancingCtx(SpatialNetDriver, this, ConnectionManager->ActorInfoMap,
+																  PrioritizedReplicationList);
 	if (LoadBalancingHandler.IsValid())
 	{
-		USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(NetDriver);
-		FSpatialReplicationGraphLoadBalancingContext LoadBalancingCtx(SpatialNetDriver, this, ConnectionManager->ActorInfoMap,
-																	  PrioritizedReplicationList);
 		LoadBalancingHandler->EvaluateActorsToMigrate(LoadBalancingCtx);
+	}
 
-		for (AActor* Actor : LoadBalancingCtx.AdditionalActorsToReplicate)
+	if (bStrategyWorkerEnabled)
+	{
+		auto& HandoverManager = SpatialNetDriver->HandoverManager;
+		TSet<Worker_EntityId_Key> ActorsToHandover = HandoverManager->GetActorsToHandover();
+		TSet<AActor*> ActorsToConsider;
```

Comment:
Could consider reserving this to ActorsToHandover size if its worth the number of allocations / wasted space trade off